### PR TITLE
implement calling of `const fn`-methods in true constants

### PR DESCRIPTION
**Pull request by oli-obk**

### CODE REVIEW

**Comment by oli-obk** [comment]
ping @arielb1

File: src/test/compile-fail/const-eval-span.rs [hunk 1]
```diff
@@ -14,7 +14,7 @@
 struct S(i32);
 
 const CONSTANT: S = S(0);
-//~^ ERROR: constant evaluation error: non-constant path in constant expression [E0080]
+//~^ ERROR: constant evaluation error: call on struct [E0080]
```
Comment (pnkfelix) [inline src/test/compile-fail/const-eval-span.rs]: (am I crazy for not understanding why this doesn't work at this point?)

Comment (pnkfelix) [inline src/test/compile-fail/const-eval-span.rs]: clarification: What I mean by that is: Is this a deliberate design decision to make this illegal?  Or is it just an artifact of the current rustc implementation?

Comment (oli-obk) [inline src/test/compile-fail/const-eval-span.rs]: just an artifact. it works in `trans/consts`, but not in `const_eval`. It's in the queue after fixing tuples and structs.

**Pull request merged**